cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -152,6 +161,19 @@ public synchronized void create(final NamespaceMeta metadata) throws Exception {
     if (!(Principal.SYSTEM.equals(principal) && NamespaceId.DEFAULT.equals(namespace))) {
       authorizationEnforcer.enforce(instanceId, principal, Action.ADMIN);
       authorizer.grant(namespace, principal, ImmutableSet.of(Action.ALL));
+      // Also grant the namespace user all privileges on the namespace, if kerberos is enabled
+      if (kerberosEnabled) {
+        ImpersonationInfo impersonationInfo = impersonationUserResolver.getImpersonationInfo(metadata);
```

Comment:
Strictly speaking, you don't need `ImpersonationUserResolver` to get the `ImpersonationInfo`. I don't like this because it introduce a circular dependency (`ImpersonationUserResolver` takes a `NamespaceQueryAdmin`, which is actually this class). All you need is a constructor in `ImpersonationInfo` that takes a `NamespaceMeta` and a `CConfiguration`. That could simplify quite some unnecessary injection of the `ImpersonationUserResolver`.